script.audio.motherearth/service.py

```diff
@@ -0,0 +1,170 @@
+        """Update the Kodi player with song metadata."""
+        song = self.last_song
+        if song and self.isPlayingAudio():
+            info = {
+                'artist': song.data['artist'],
+                'title': song.data['title'],
+                'genre': '',
+            }
+            if 'album' in song.data:
+                info['album'] = song.data['album']
+            if 'rating' in song.data:
+                rating = float(song.data['rating'])
+                info['rating'] = rating
+                info['userrating'] = int(round(rating))
+            if 'year' in song.data:
+                info['year'] = int(song.data['year'])
+            item = xbmcgui.ListItem()
+            item.setPath(self.getPlayingFile())
+            item.setArt({'thumb': song.cover})
+            item.setArt({'fanart': song.fanart})
+            item.setInfo('music', info)
+            self.updateInfoTag(item)
+
+    def update_song(self):
+        """Update song metadata, if necessary."""
+        last_key = self.last_key
+        last_song = self.last_song
+        song_key = self.get_song_key()
+        if song_key is None or song_key == last_key:
+            return
+
+        if song_key != ('', ''):
+            xbmc.log(f'rp_service: song_key {song_key}', xbmc.LOGDEBUG)
+            song_data = self.now_playing.get_song_data(song_key)
+            if song_data:
+                self.last_key = song_key
+        else:
+            song_data = self.now_playing.current
+        if song_data is None or last_song and last_song.data == song_data:
+            return
+
+        cover = song_data.get('cover')
+        xbmc.log(f'me_service: cover {cover}', xbmc.LOGDEBUG)
+
+        addon = xbmcaddon.Addon()
+
+        self.last_song = Song(song_data, cover, fanart)
+        self.update_player()
+
+    def onAVStarted(self):
+        if self.isPlaying() and self.getPlayingFile() in STREAM_INFO:
+            url = self.getPlayingFile()
+            info = STREAM_INFO[url]
+            # Kodi switches to fullscreen for FLAC, but not AAC
+            if url == info['url_aac']:
+                xbmc.executebuiltin('Action(FullScreen)')
+        else:
+            self.reset()
+
+    def onPlayBackEnded(self):
+        if self.stream_url:
+            self.restart_time = time.time()
+        else:
+            self.reset()
+
+    def onPlayBackError(self):
+        self.reset()
+
+    def onPlayBackStarted(self):
+        if self.isPlaying() and self.getPlayingFile() in STREAM_INFO:
+            url = self.getPlayingFile()
+            self.stream_url = url
+            self.restart_time = 0
+            info = STREAM_INFO[url]
+            self.now_playing.set_channel(info['channel'])
+        else:
+            self.reset()
+
+    def onPlayBackStopped(self):
+        self.reset()
+
+
+
+if __name__ == '__main__':
+    player = Player()
+    monitor = xbmc.Monitor()
+    while not monitor.abortRequested():
+        if monitor.waitForAbort(0.1):
```

Comment:
any reason for such a small sleep time? 1 second would be enough no?

Reply:
please also check addon-checker errors and warnings:

```
WARN: http://github.com/FlorianReiterer/script.audio.motherearth redirects to https://github.com/FlorianReiterer/script.audio.motherearth
INFO: Image icon exists
ERROR: icon should be solid. It has transparency.
ERROR: icon should have either 256x256 or 512x512 but it has 300x300
INFO: Image fanart exists
ERROR: fanart should have either 1280x720 or 1920x1080 or 3840x2160 but it has 1024x1025
INFO: fanart file size is fine 289KB
WARN: Complex entry point. Check: script.py | Counted lines: 48 | Lines allowed: 15
WARN: Complex entry point. Check: service.py | Counted lines: 139 | Lines allowed: 15
```

Reply:
fixed the artwork, and changed the sleep time to 1:
`        if monitor.waitForAbort(1):`

struggle with the http / https github redirect, sorry, will sort that out.
also i try shortening the script.py: i move the contents of it to another python script file and just point script.py to there like:
```
import new.py
new.run()
```
 right?
..but i struggle with that also. would you mind keeping those extra lines for now, i will get me some help and fix this as soon as i can.

Reply:
the entrypoint size warning is just a recomendation, not something mandatory. What happens is that kodi does not compile the entrypoint to bytecode, so it compiles it every time the plugin runs. If you store most of it on a module and just execute (like above), the ammount of code to compile is far smaller. But you can do this later or even ignore.

Regarding the redirect to https, please use the https link in your adon.xml (`<source>` element).

Reply:
thanks! https is fixed also now, i'll go with the longer code for a start.